src/lib/datepicker/datepicker-input.ts

```diff
@@ -123,38 +123,37 @@ export class MatDatepickerInput<D> implements ControlValueAccessor, OnDestroy, V
 
   /** The value of the input. */
   @Input()
-  get value(): D | null { return this._value; }
+  get value(): D | null { return this._value? this._value.asDate(): null; }
   set value(value: D | null) {
-    value = this._dateAdapter.deserialize(value);
-    this._lastValueValid = !value || this._dateAdapter.isValid(value);
-    value = this._getValidDateOrNull(value);
-    const oldDate = this.value;
-    this._value = value;
-    this._formatValue(value);
-
-    if (!this._dateAdapter.sameDate(oldDate, value)) {
-      this._valueChange.emit(value);
+    const oldDate = this._value;
+    this._value = new MatSingleDateSelection(this._dateAdapter, value);
```

Comment:
I think we decided to have a single instance of `MatSingleDateSelection` for the lifetime of the input component. Instead of creating a new one we can set the selected value. There's also an observable on `MatDateSelection` that you can subscribe to if you need to know when it changes

Reply:
you mean there should be an observable on matdateselection?

Reply:
oh i found it hadn't sync'd